app.py

```diff
@@ -175,5 +194,42 @@ def extra4():
     """extra route to work with"""
     return
 
+@app.route("/save_fav_song", methods=["GET", "POST"])
+def save_fav_song():
+    """allows user to save a song on the favorites page"""
+    fav_song_name = flask.request.form.get("favSong")
+    fav_song = favSongs(song=fav_song_name, username=current_user.username)
+    db.session.add(fav_song)
+    db.session.commit()
+    return flask.redirect("/favorites")
+
+@app.route("/del_fav_song", methods=["GET", "POST"])
+def del_fav_song():
+    """allows user to delete a song on the favorites page"""
+    fav_song_name = flask.request.form.get("favSong")
+    fav_song = favSongs.query.filter_by(song=fav_song_name).first()
```

Comment:
You will need to query the song and then check if the song in the database that was queried belongs to that user. for example, right now if you deleted a favorite song that belongs to you, but it was also my favorite song, we don't know who's song would be deleted. so you will need to cross check the song title and the username in the database. same thing for the del_fav_artist